### PR TITLE
logged in users can now see discoverable resources

### PR DESCRIPTION
**Pull request by pkdash**

### CODE REVIEW

**Comment by ironfroggy** [comment]
+1

File: hs_core/hydroshare/users.py [hunk 1]
```diff
@@ -707,14 +705,16 @@ def _filter_resources_for_user_and_owner(user, owner, is_editable, query, is_pub
                 query.append(Q(pk__in=owner.uaccess.get_owned_resources()))
 
                 if user != owner:
-                    is_public = True
+                    # if some authenticated user is asking for resources owned by another user then
+                    # get other user's owned resources that are public or discoverable
+                    query.append(Q(raccess__public=True) | Q(raccess__discoverable=True))
```
Comment (hyi) [inline hs_core/hydroshare/users.py]: @pkdash I am thinking whether we need to add access control check here, i.e., if some authenticated user is asking for resources owned by another user, in addition to querying public and discoverable resources, whether private resources this particular user has view privileges from the owner should also be visible for this user?

Comment (pkdash) [inline hs_core/hydroshare/users.py]: @hyi I think you are right. We should include private resources if the requesting user has access to it.

**Comment by hyi** [comment]
+1

**Pull request merged**